src/lib/history.ts

```diff
@@ -52,17 +60,46 @@ export default class HistoryHandler {
     this.updateState();
   };
 
+  /**
+   * Have history remember the current selection {@param objects},
+   * in case it is deleted/modified.
+   */
   store = (objects: readonly fabric.Object[]): void => {
     if (this.locked) return;
     this.locked = true;
     this.selection = this.canvas.serialize(objects);
     this.locked = false;
   };
 
+  /**
+   * If the active selection (known to history) is modified to become {@param objects}
+   */
   modify = (objects: fabric.Object[]): void =>
     this.save({ oldObjects: this.selection, newObjects: objects });
 
-  save = ({
+  private getNextId = (): number => {
+    return ++this.latestId;
+  };
+
+  private setIdIfNotPresent = (object: FabricObject): number => {
+    AssertType<ObjectId>(object);
+    if (object.idVersion !== 1 || object.id == null) {
```

Comment:
why a null check as opposed to `=== undefined`?

Reply:
In case it is also null; we don't fully trust this external data so this may help

(Also, simply because we're trying to check whether a field *exists*, and this is the common idiom for that)

Reply:
i disagree that `== null` is cleaner or more idiomatic. i've seen `=== undefined` appear in library code, i've never seen `== null` before. our linter also doesn't like it

Reply:
I'm not a fan of using `=== undefined` to check whether something exists, but the change has been made